internal/pkg/manifest/workload.go

```diff
@@ -74,6 +76,31 @@ type Workload struct {
 	Type *string `yaml:"type"` // must be one of the supported manifest types.
 }
 
+// OverrideRule holds the manifest overriding rule for CloudFormation template.
+type OverrideRule struct {
+	TaskDefOverrideRules []TaskDefinitionOverrideRule `yaml:"taskdef_override"`
+}
+
+// TaskDefinitionOverrideRule holds the overriding rule for ECS Task Definition.
+type TaskDefinitionOverrideRule string
+
+// Parse parses the manifest ECS Task Definition overriding rule into override.Rule.
+func (r TaskDefinitionOverrideRule) Parse() (*override.Rule, error) {
+	// Make sure the overriding rule is in a map format.
+	// For example: "ContainerDefinitions[0].Ulimits[-].HardLimit: !Ref ParamName"
+	if len(strings.Split(string(r), ":")) != 2 {
```

Comment:
hmmm instead of a single string that contains both key and value, I think we should aim to provide a similar experience to CloudFormation intrinsic functions.


```yaml
# [preferred] Option 1: explicit values
taskdef_override:
- 
   path: 'ContainerDefinitions[0].Ulimits[-].HardLimit'
   value: !Ref ParamName

# Option 2: map[string]*yaml.Node
taskdef_override:
- "ContainerDefinitions[0].Ulimits[-].HardLimit": !Ref ParamName
```

I know we discussed this before 💭 , but I think I'm leaning towards option 1 because it felt odd to type option 2 since it doesn't have a specific structure.